### ISSUE TITLE
Unexpected result

### ISSUE THREAD

**Issue by derrickpelletier** [mikolalysenko/split-polygon#1]
I seem to be getting an unexpected split.
Though I'm not particularly familiar with planes, my understanding is that a plane such as `[0, 1, -50]` would be a plane running along N 50° (the negative threw me off, but that seems to be how it's working).

Given a polygon: 
`[[-122.350051, 47.702893 ], [-122.344774, 47.702877 ], [-122.344777, 47.70324 ], [-122.341982, 47.703234 ], [-122.341959, 47.701421 ], [-122.339749, 47.701416 ], [-122.339704, 47.69776 ], [-122.341913, 47.697797 ], [-122.341905, 47.697071 ], [-122.344576, 47.697084 ], [-122.344609, 47.697807 ], [-122.349999, 47.697822 ], [-122.350051, 47.702893 ]]`
And a plane:
`[0, 1, -47.6971435546875]`

This is the resulting split I get.
http://geojson.io/#id=gist:anonymous/536b911803935ddeffe2&map=16/47.7002/-122.3449

I've split this sequentially on intervals starting at the top, and each iteration split perfectly horizontal until this one, so unless I'm missing something, there's a bug.

Would have tried to fix it, but the code is pretty terse and undocumented so I'm not sure what's going on exactly.

---

updated geojson.io with a line representing plane.

**Comment by mikolalysenko** [comment]
Hmm..  That is weird.

Also, I just realized that the documentation for this module is wrong.  The input polygon needs to be convex in order for the splitting to be correct.  Though that isn't your problem here, I will go back and try to fix it.

**Comment by derrickpelletier** [comment]
Ah shoot, I didn't realize that it was only for convex.
That poly I posted is concave, though, so I suppose it could have something to do with the bug?

**Comment by mikolalysenko** [comment]
I think that there were actually two bugs.

The first was that there was a problem with the way it was intersecting line segments with planes, and the second was that the documentation was wrong.

Unfortunately though, I think the second issue may rule out you using this module for your application unless you can split it into convex pieces :(

**Comment by derrickpelletier** [comment]
Yeah, that's unfortunate! Such a cleaner usage pattern than some of the other alternatives too.
Thanks for the quick response, regardless.
